planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -47,74 +51,64 @@ public Itinerary generateItinerary(List<Event> events) {
-    // The order of this graph will be the same as the order of the events.
-    // For the MVP, real time traffic is NOT used.
-    // TODO: handle invalid addresses
-    private Duration[][] getEventsDistanceMatrix(List<Event> events) {
-        int numOfNodes = events.size();
-        Duration[][] travelTimeGraph = new Duration[numOfNodes][numOfNodes];
-
-        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
-        String[] origins = getListOfAddresses(events);
-        String[] destinations = getListOfAddresses(events);
-        GeoApiContext context = new GeoApiContext.Builder(new GaeRequestHandler.Builder())
-                            .apiKey(GoogleApiKey)
-                            .build();
-       try {
-            DistanceMatrix distanceMatrix = DistanceMatrixApi.getDistanceMatrix(context,
-                                                         origins,
-                                                         destinations).await();
+    // Function that creates an itinerary by scheduling each event in order.
+    private List<ItineraryItem> scheduleItineraryInOrder(List<Event> events, int openningTime, int endingTime) {
+        List<ItineraryItem> items = new ArrayList<>();
+        boolean optimized = false;
+        DirectionsRoute directionsRoute = getDirectionsRoute(events, optimized);
+        if (errorMessage != null) { 
+            return items;
+        }
+        DirectionsLeg[] directionsLegs = directionsRoute.legs;
+        int start = openningTime; 
+        for (int i = 0; i < events.size(); i++){
+            Event event = events.get(i);
+            // Add the event as an itinerary item if the remaining available time is longer than the
+            // event duration.
+            if (start + event.getDurationInMinutes() <= endingTime) {
+                ItineraryItem item = new ItineraryItem(event.getName(), event.getAddress(),
+                    TimeRange.fromStartDuration(start, event.getDurationInMinutes()));
+                items.add(item);
 
-            for (int i = 0; i < distanceMatrix.rows.length; i++) {
-                DistanceMatrixRow currentRow = distanceMatrix.rows[i];
-                for (int j = 0; j < currentRow.elements.length; j++) {
-                    DistanceMatrixElement currentElement = currentRow.elements[j];
-                    if (currentElement.duration != null) {
-                        travelTimeGraph[i][j] = currentElement.duration;
-                    } else {
-                        errorMessage = "Oops, one of your addresses is invalid, please use a valid address.";
-                    }
-                }
+                // Update the next event's start time
+                start += event.getDurationInMinutes() + getTravelDurationInMinutes(directionsLegs[i].duration);
+            }else {
+                errorMessage = "Sorry, you have too many events in a day!";
+                return Arrays.asList();
```

Comment:
Is it possible to use the same code for both this and for the optimized loop? I feel like waypointorder would still work correctly if optimized is false (it will just be [0..N]). Or does it that not work?

Reply:
the waypoint order would be an empty array if optimized is false

Reply:
I can pass optimized as a parameter and set eventIndex to i if optimized is false, but that might make the code less readable